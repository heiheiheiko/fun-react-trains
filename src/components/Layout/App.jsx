import React from 'react';
import { BrowserRouter as Router, Route, NavLink } from "react-router-dom";
import pages from "../../config/Pages";
import AppAside from "./AppAside";

function App(){
  return (
    <Router>
      <div className="overlay">
        <nav className="overlay__layer mt-small py-small px-normal has-background-light">
          {pages.map((navItem) =>
            <NavLink 
              className="ml-normal"
              to={navItem.path} 
              key={navItem.id} 
              activeClassName="has-text-primary"> 
              {navItem.title} 
            </NavLink>  
          )}
        </nav>
        
        {pages.map((navItem) =>
          <Route path={navItem.path} render={() => navItem.component} key={navItem.id}/>
        )}

        <AppAside/>
      </div>
    </Router>
  );
}

export default App;
