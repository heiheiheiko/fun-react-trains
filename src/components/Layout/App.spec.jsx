import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from "react-redux";
import renderer from 'react-test-renderer'

import App from './App';

jest.mock('./AppAside', () => ()=> <div> AppAside </div>);

describe('<App>', () => {
  const initialState = { mapCenter: [1,1] };
  const mockStore = configureStore();
  let store, componentContainer;

  beforeEach(() => {
    store = mockStore(initialState)
    componentContainer = <Provider store={store}><App /></Provider>;
  })

  it('could be renderd', () => {
    expect(mount(componentContainer).find(App).length).toEqual(1)
  });

  it('matched the snapshot', () => {
    const renderedValue =  renderer.create(componentContainer).toJSON()
    expect(renderedValue).toMatchSnapshot();
  });
});