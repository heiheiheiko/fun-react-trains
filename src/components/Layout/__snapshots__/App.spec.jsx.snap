// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> matched the snapshot 1`] = `
<div
  className="overlay"
>
  <nav
    className="overlay__layer mt-small py-small px-normal has-background-light"
  >
    <a
      aria-current={null}
      className="ml-normal"
      href="/world"
      onClick={[Function]}
    >
      World
    </a>
    <a
      aria-current={null}
      className="ml-normal"
      href="/finnland"
      onClick={[Function]}
    >
      Finnland
    </a>
  </nav>
  <div>
     AppAside 
  </div>
</div>
`;
